include/zephyr/toolchain/gcc.h

```diff
@@ -14,6 +14,7 @@
  * Macros to abstract compiler capabilities for GCC toolchain.
  */
 
+#undef GCC_VERSION	/* clear out any existing definition */
```

Comment:
Per my grepping, this symbol is only ever used in this one file.  So if there's a collision why don't we just rename it?  Stick an underscore on it or something?

Reply:
or even a ZEPHYR_ prefix? Sounds like a better solution.

Reply:
and CLANG_VERSION too; avoids potential trouble with un-namespaced symbols.